pkg/codegen/python/gen.go

```diff
@@ -136,7 +136,15 @@ func (mod *modContext) gen(fs fs) error {
 			d = ""
 		}
 		if d == dir {
-			exports = append(exports, p[:len(p)-len(".py")])
+			// this ensures that we can look at nested files
+			// without this, we don't understand where the destFiles are
+			rawNameParts := strings.Split(p, "/")
+			if len(rawNameParts) == 1 {
+				exports = append(exports, p[:len(p)-len(".py")])
+			} else {
+				fileName := rawNameParts[len(rawNameParts)-1]
+				exports = append(exports, fileName[:len(fileName)-len(".py")])
+			}
```

Comment:
I suspect this won't work on Windows (`"/"`).  But if I'm reading it right, all of this code can be replaced with:

```go
exports = append(exports, strings.TrimSuffix(path.Base(p), ".py"))
```

Reply:
nice suggestion - pushed! will merge on green